### PR TITLE
[Upgrade] Refine Sylius 1.14 Upgrade Guide

### PR DESCRIPTION
**Pull request by Rafikooo**
| Q               | A
|-----------------|-----
| Branch?         | 1.14  
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no 
| License         | MIT

### CODE REVIEW

File: UPGRADE-1.14.md [hunk 1]
```diff
@@ -75,6 +75,7 @@
     | `sylius.event_subscriber.admin_cache_control_subscriber`                                                            | `sylius_admin.event_subscriber.admin_section_cache_control`                                         |
     | `sylius.event_subscriber.admin_filter_subscriber`                                                                   | `sylius_admin.event_subscriber.admin_filter`                                                        |
     | `sylius.admin.menu_builder.main`                                                                                    | `sylius_admin.menu_builder.main`                                                                    |
+    | `sylius.admin.main`                                                                                                 | `sylius_admin.main`                                                                                 |
```
Comment (GSadee) [inline UPGRADE-1.14.md]: It is an alias for a tag, not for a service
```suggestion
```

File: UPGRADE-1.14.md [hunk 2]
```diff
@@ -529,6 +530,7 @@
     | `sylius.listener.user_cart_recalculation`                                                                           | `sylius_shop.listener.user_cart_recalculation`                                                      |
     | `sylius.listener.user_impersonated`                                                                                 | `sylius_shop.listener.user_impersonated`                                                            |
     | `sylius.shop.menu_builder.account`                                                                                  | `sylius_shop.menu_builder.account`                                                                  |
+    | `sylius.shop.account`                                                                                               | `sylius_shop.account`                                                                               |
```
Comment (GSadee) [inline UPGRADE-1.14.md]: It is an alias for a tag, not for a service
```suggestion
```

File: UPGRADE-1.14.md [hunk 3]
```diff
@@ -619,16 +621,7 @@
...
-1. Aliases for the following `knp_menu.menu_builder` service tags have been introduced to standardize tag aliases and will replace the incorrect aliases in Sylius 2.0:
-
-    | Old Alias             | New Alias             |
-    |-----------------------|-----------------------|
-    | **AdminBundle**       |                       |
-    | `sylius.admin.main`   | `sylius_admin.main`   |
-    | **ShopBundle**        |                       |
-    | `sylius.shop.account` | `sylius_shop.account` |
-
-    The old alias are now deprecated and will be removed in Sylius 2.0.
```
Comment (GSadee) [inline UPGRADE-1.14.md]: This should be restored

File: UPGRADE-1.14.md [hunk 4]
```diff
@@ -639,98 +632,97 @@
...
-    - `Sylius\Bundle\PromotionBundle\Validator\Constraints\CatalogPromotionScope` 
-   Use the regular Symfony validation constraints instead.
-
-1. The class `Sylius\Bundle\CoreBundle\Twig\StateMachineExtension` has been deprecated and will be removed in Sylius 2.0. Use `Sylius\Abstraction\StateMachine\Twig\StateMachineExtension` instead.
-
-1. The class `Sylius\Bundle\CoreBundle\Console\Command\ShowAvailablePluginsCommand` has been deprecated and will be removed in Sylius 2.0.
+## Classes and Interfaces Deprecations
 
-1. The class `Sylius\Bundle\CoreBundle\Console\Command\Model\PluginInfo` has been deprecated and will be removed in Sylius 2.0.
+| Class/Interface                                                       | Replacement                                                             |
```
Comment (GSadee) [inline UPGRADE-1.14.md]:
```suggestion
| Deprecated class/interface                                                       | Will be replaced with                                                              |
```
🤔

File: UPGRADE-1.14.md [hunk 5]
```diff
@@ -639,98 +632,97 @@
...
+|-----------------------------------------------------------------------|-------------------------------------------------------------------------|
+| `Sylius\Bundle\AdminBundle\Provider\StatisticsDataProvider`           | `Sylius\Component\Core\Statistics\Provider\StatisticsProvider`          |
+| `Sylius\Bundle\AdminBundle\Provider\StatisticsDataProviderInterface`  | `Sylius\Component\Core\Statistics\Provider\StatisticsProviderInterface` |
+| `Sylius\Bundle\CoreBundle\Twig\StateMachineExtension`                 | `Sylius\Abstraction\StateMachine\Twig\StateMachineExtension`            |
 
-1. The class `Sylius\Bundle\CoreBundle\Form\EventSubscriber\AddUserFormSubscriber` has been deprecated and will be removed in Sylius 2.0.
+| Class/Interface                                                       | Will be replaced with                                                   |
+|-----------------------------------------------------------------------|-------------------------------------------------------------------------|
```
Comment (GSadee) [inline UPGRADE-1.14.md]:
```suggestion
```

File: UPGRADE-1.14.md [hunk 6]
```diff
@@ -639,98 +632,97 @@
...
-    - `Sylius\Bundle\ApiBundle\Exception\ProductVariantCannotBeRemoved`
-    - `Sylius\Bundle\ApiBundle\Exception\PromotionCannotBeRemoved`
-    - `Sylius\Bundle\ApiBundle\Exception\PromotionCouponCannotBeRemoved`
-    - `Sylius\Bundle\ApiBundle\Exception\ProvinceCannotBeRemoved`
-    - `Sylius\Bundle\ApiBundle\Exception\ShippingMethodCannotBeRemoved`
-    - `Sylius\Bundle\ApiBundle\Exception\TaxonCannotBeRemoved`
-    - `Sylius\Bundle\ApiBundle\Exception\ZoneCannotBeRemoved`
-
-    All of these exceptions will be replaced with the `Sylius\Component\Core\Exception\ResourceDeleteException` exception.
+| Class/Interface                                                                                                        |
```
Comment (GSadee) [inline UPGRADE-1.14.md]: As in this table is only a single column, I would be probably for not using a table here 🤔

File: UPGRADE-1.14.md [hunk 7]
```diff
@@ -639,98 +632,97 @@
...
-            private RequestStack|SessionInterface $requestStackOrSession,
-    +       private ?RouterInterface $router = null,
-        )
-    ```
-
 1. The following classes have been deprecated and will be removed in Sylius 2.0:
     - **AddressingBundle**
     - `Sylius\Bundle\AddressingBundle\Controller\ProvinceController`
     - **AdminBundle**
     - `Sylius\Bundle\AdminBundle\Controller\NotificationController`
```
Comment (GSadee) [inline UPGRADE-1.14.md]: Why are they not above with the other classes?

File: UPGRADE-1.14.md (hunk 6 above)
Comment (NoResponseMate) [reply UPGRADE-1.14.md]: :+1: a simple list would be better

File: UPGRADE-1.14.md [hunk 8]
```diff
@@ -1036,6 +1032,10 @@
     * `sylius.{user_type}_user.pin_generator.password_reset`
     * `sylius.{user_type}_user.pin_uniqueness_checker.password_reset`
 
+## Miscellaneous
+
+1. Starting with Sylius 1.14, form types will be extended using the parent form instead of through form extensions, like it's done in the `Sylius\Bundle\AdminBundle\Form\Type\CatalogPromotionScopeType` and `Sylius\Bundle\AdminBundle\Form\Type\CatalogPromotionActionType` classes.
+
```
Comment (NoResponseMate) [inline UPGRADE-1.14.md]:
```suggestion
## Miscellaneous

1. Starting with Sylius 1.14, form types will be extended using the parent form instead of through form extensions, like it's done in the `Sylius\Bundle\AdminBundle\Form\Type\CatalogPromotionScopeType` and `Sylius\Bundle\AdminBundle\Form\Type\CatalogPromotionActionType` classes.

```
IMO there's too little info in here.
Also don't we have, or rather should, an architecture changes section?

File: src/Sylius/Bundle/ApiBundle/PropertyInfo/Extractor/EmptyPropertyListExtractor.php [hunk 9]
```diff
@@ -15,6 +15,14 @@
...
 use Symfony\Component\PropertyInfo\PropertyListExtractorInterface;
 
+trigger_deprecation(
+    'sylius/api-bundle',
+    '1.14',
+    'The "%s" class is deprecated and will be removed in Sylius 2.0.',
+    EmptyPropertyListExtractor::class,
+);
+
+/** @deprecated since Sylius 1.14 and will be removed in Sylius 2.0. */
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/PropertyInfo/Extractor/EmptyPropertyListExtractor.php]: This service has not yet been removed in 2.0, so it is already too late to deprecate and remove it

File: src/Sylius/Bundle/ApiBundle/Resources/config/services.xml [hunk 10]
```diff
@@ -32,8 +32,11 @@
 
         <service id="Sylius\Bundle\ApiBundle\PropertyInfo\Extractor\EmptyPropertyListExtractor">
             <tag name="property_info.list_extractor" priority="-2000" />
+            <deprecated package="sylius/api-bundle" version="1.14">The "%service_id%" service is deprecated since 1.14 and will be removed in 2.0.</deprecated>
+        </service>
+        <service id="sylius_api.extractor.property_info.empty_property_list" alias="Sylius\Bundle\ApiBundle\PropertyInfo\Extractor\EmptyPropertyListExtractor">
+            <deprecated package="sylius/api-bundle" version="1.14">The "%alias_id%" service is deprecated since 1.14 and will be removed in 2.0.</deprecated>
         </service>
-        <service id="sylius_api.extractor.property_info.empty_property_list" alias="Sylius\Bundle\ApiBundle\PropertyInfo\Extractor\EmptyPropertyListExtractor" />
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services.xml]: Revert this changes according to the above comment

File: UPGRADE-1.14.md [hunk 11]
```diff
@@ -188,7 +188,6 @@
     | `Sylius\Bundle\ApiBundle\Validator\Constraints\UniqueReviewerEmailValidator`                                        | `sylius_api.validator.unique_reviewer_email`                                                        |
     | `Sylius\Bundle\ApiBundle\Validator\Constraints\AdminResetPasswordTokenNonExpiredValidator`                          | `sylius_api.validator.admin_reset_password_token_non_expired`                                       |
     | `sylius.validator.order_address_requirement`                                                                        | `sylius_api.validator.order_address_requirement`                                                    |
-    | `Sylius\Bundle\ApiBundle\PropertyInfo\Extractor\EmptyPropertyListExtractor`                                         | `sylius_api.extractor.property_info.empty_property_list`                                            |
```
Comment (GSadee) [inline UPGRADE-1.14.md]: It should be reverted

**Review by GSadee — APPROVED** [review]
